### PR TITLE
DROOLS-2359 DMN DecisionTable output validation when type is collection

### PR DESCRIPTION
**Pull request by tarilabs**

### CODE REVIEW

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNEvaluatorCompiler.java [hunk 1]
```diff
@@ -455,6 +455,12 @@ private DMNExpressionEvaluator compileDecisionTable(DMNCompilerContext ctx, DMNM
                 if( typeRef == null ) {
                     typeRef = (BaseDMNTypeImpl) DMNTypeRegistry.UNKNOWN;
                 }
+            } else if (dt.getOutput().size() == 1 && (dt.getParent() instanceof Decision || dt.getParent() instanceof BusinessKnowledgeModel || dt.getParent() instanceof ContextEntry)) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNEvaluatorCompiler.java:458]: moving type inference up, before understanding which List of Value to consider for validation of the DT output.
Type inference should work regardless.

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ public void testSimpleItemDefinition() {
         context.set( "Monthly Salary", 1000 );
 
         DMNResult dmnResult = runtime.evaluateAll( dmnModel, context );
-        assertThat( DMNRuntimeUtil.formatMessages( dmnResult.getMessages() ), dmnModel.hasErrors(), is( false ) );
+        assertThat(DMNRuntimeUtil.formatMessages(dmnResult.getMessages()), dmnResult.hasErrors(), is(false));
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java:96]: typo fix.

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java [hunk 3]
```diff
@@ -1805,10 +1805,90 @@ public void testEnhancedForLoop() {
         DMNContext emptyContext = DMNFactory.newContext();
 
         DMNResult dmnResult = runtime.evaluateAll(dmnModel, emptyContext);
-        assertThat(DMNRuntimeUtil.formatMessages(dmnResult.getMessages()), dmnModel.hasErrors(), is(false));
+        assertThat(DMNRuntimeUtil.formatMessages(dmnResult.getMessages()), dmnResult.hasErrors(), is(false));
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java:1808]: typo fix.

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java [hunk 4]
```diff
@@ -1805,10 +1805,90 @@ public void testEnhancedForLoop() {
...
 
         DMNContext result = dmnResult.getContext();
         assertThat(result.get("an hardcoded forloop"), is(Arrays.asList(new BigDecimal(2), new BigDecimal(3), new BigDecimal(4))));
     }
+
+    @Test
+    public void testDecisionTableOutputDMNTypeCollection() {
```
Comment (baldimir) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: Nitpicking: There is a DMNDecisionTableRuntimeTest class. I think these tests can go there. 

https://github.com/baldimir/drools/blob/af579f5e65f724fb836b449602c8d30be81c7611/kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNDecisionTableRuntimeTest.java#L50

**Review by baldimir — APPROVED** [review]

**Review by etirelli — APPROVED** [review]

**Pull request merged**